File: 651A.py
x,y = map(int, input().split())

if x == 1 and y ==1:
    print(0)
else:
    a = max(x, y)
    b = min(x, y)

    c = 0
    while a != 0 and b != 0:
        a -= 2
        b += 1
        if a <= 2:
            a, b = b, a
        c += 1

    print(c)

File: 651A.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
    int a,b,x,y;
    cin>>x>>y;
    if(x==1 && y==1){
        cout<<"0"<<endl;
        return 0;
    }
    a=max(x,y);
    b=min(x,y);

    int c=0;
    while(a!=0 && b!=0){
        a-=2;
        b++;
        if(a<=2){
            swap(a,b);
        }

        c++;
    }

    cout<<c<<endl;
}